.devcontainer/Dockerfile

```diff
@@ -0,0 +1,100 @@
+
+# [Choice] Python version: 3, 3.9, 3.8, 3.7, 3.6
+ARG VARIANT="3.8"
+FROM mcr.microsoft.com/vscode/devcontainers/python:dev-${VARIANT}-buster
+
+# This will be set to true when running in VSCode
+ARG INTERACTIVE="false"
+
+ARG USERNAME=vscode
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+# Set up non-root user
+COPY .devcontainer/scripts/non-root-user.sh /tmp/
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN bash /tmp/non-root-user.sh "${USERNAME}" "${USER_UID}" "${USER_GID}"
+
+# Set env for tracking that we're running in a devcontainer
+ENV DEVCONTAINER=true
+
+# Install Node.js for GH actions tests and UI
+ARG NODE_VERSION="lts/*"
+RUN su $USERNAME -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"
+
+# Install terraform
+ARG TERRAFORM_VERSION="1.1.5"
+COPY .devcontainer/scripts/terraform.sh /tmp/
+RUN bash /tmp/terraform.sh "${TERRAFORM_VERSION}" /usr/bin
+
+ARG DOCKER_GROUP_ID
+COPY .devcontainer/scripts/docker-client.sh /tmp/
+RUN /tmp/docker-client.sh $USERNAME
+
+# Install Docker
+# hadolint ignore=DL3008,DL3004
+RUN apt-get update && sudo apt-get install -y apt-transport-https ca-certificates curl gnupg lsb-release \
+    && curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+    && echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/debian $(lsb_release -cs) stable" \
+    | tee /etc/apt/sources.list.d/docker.list > /dev/null \
+    && apt-get update && apt-get install -y docker-ce docker-ce-cli containerd.io \
+    && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+# Install nekos act - run GitHub workflows locally https://github.com/nektos/act
+# hadolint ignore=DL3004
+RUN if [ "${INTERACTIVE}" = "true" ]; then curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash; fi
+
+# Install Certbot
+# hadolint ignore=DL3008
+RUN apt-get update && apt-get install -y python3 python3-venv libaugeas0 \
+    && python3 -m venv /opt/certbot/ \
+    && /opt/certbot/bin/pip install --no-cache-dir --upgrade pip \
+    && /opt/certbot/bin/pip install --no-cache-dir certbot \
+    && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+# Install Porter
+# Not using the script from https://cdn.porter.sh/latest/install-linux.sh
+# as it installs things we don't need and duplicates the binary.
+ARG PORTER_MIRROR=https://cdn.porter.sh
+ARG PORTER_PERMALINK=v0.38.11
+ARG PORTER_PKG_PERMALINK=latest
+ARG PORTER_HOME=/home/$USERNAME/.porter/
+COPY .devcontainer/scripts/porter.sh /tmp/
+RUN export PORTER_MIRROR=${PORTER_MIRROR} \
+    PORTER_PERMALINK=${PORTER_PERMALINK} \
+    PORTER_PKG_PERMALINK=${PORTER_PKG_PERMALINK} \
+    PORTER_HOME=${PORTER_HOME} \
+    && /tmp/porter.sh
+
+ENV PATH ${PORTER_HOME}:$PATH
+
+# Install azure-cli
+ARG AZURE_CLI_VERSION=2.36.0-1~buster
```

Comment:
The auth scripts in AzureTRE have recently been updated, and AFAIK now require 2.37 or later of `az`

Reply:
updated